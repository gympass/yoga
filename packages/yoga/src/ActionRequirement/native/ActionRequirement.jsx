import React from 'react';
import { arrayOf, bool, node, oneOfType, string } from 'prop-types';
import styled from 'styled-components/native';
import { View } from 'react-native';
import Box from '../../Box';
import { Title } from './ActionRequirementStyles';
import Text from '../../Text';

const StyledActionRequirement = styled.View``;

function ActionRequirement(props) {
  const {
    title,
    description,
    children,
    checkable,
    illustration,
    list,
    titleAsTextDisplay,
  } = props;

  return (
    <StyledActionRequirement {...props}>
      {illustration && <Box>{illustration}</Box>}
      {titleAsTextDisplay ? (
        <Text.Display2>{title}</Text.Display2>
      ) : (
        <Title>{title}</Title>
      )}
      <Text.Body1 mt="small" color="deep">
        {description}
      </Text.Body1>
      {list && (
        <Text.Body1 mt="small" color="deep">
          <Box mt="xxxlarge">{list}</Box>
        </Text.Body1>
      )}
      {checkable && <Box mt="xxxlarge">{checkable}</Box>}
      <View>{children}</View>
    </StyledActionRequirement>
  );
}

ActionRequirement.propTypes = {
  title: oneOfType([arrayOf(node), node]).isRequired,
  children: oneOfType([arrayOf(node), node]),
  description: string.isRequired,
  checkable: oneOfType([arrayOf(node), node]),
  illustration: oneOfType([arrayOf(node), node]),
  list: oneOfType([arrayOf(node), node]),
  titleAsTextDisplay: bool,
};

ActionRequirement.defaultProps = {
  children: undefined,
  checkable: undefined,
  illustration: undefined,
  list: undefined,
  titleAsTextDisplay: false,
};

export default ActionRequirement;
