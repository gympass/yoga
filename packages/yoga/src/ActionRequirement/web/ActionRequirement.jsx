import React from 'react';
import { arrayOf, bool, node, oneOfType, string } from 'prop-types';
import styled from 'styled-components';
import { media } from '@gympass/yoga-helpers';
import {
  Actions,
  Title,
  PrimaryButton,
  SecondaryButton,
} from './ActionRequirementStyles';
import Text from '../../Text';
import Box from '../../Box';

const StyledActionRequirement = styled.div`
  display: flex;
  height: 100%;
  ${media.xxs`
    flex-direction: column;
  `}
  ${media.lg`
    flex-direction: row-reverse;
  `}
`;

const Content = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;

  ${media.xxs`
    width: auto;
    margin-right: 0;
  `}

  ${media.lg`
    width: 520px;
    margin-right: 120px;
  `}
`;

const BoxIllustration = styled(Box)`
  flex-grow: 1;
  display: flex;
  justify-content: center;
  align-items: center;
`;

function isChildFromComponent(child, component) {
  return child.type.displayName === component.displayName;
}

function ActionRequirement(props) {
  const {
    title,
    description,
    children,
    checkable,
    illustration,
    list,
    titleAsTextDisplay = false,
    titleLevel = 'h1',
  } = props;

  let primaryButton;
  let secondaryButton;

  function defineCompoundComponents() {
    React.Children.forEach(children, child => {
      if (isChildFromComponent(child, PrimaryButton)) primaryButton = child;
      if (isChildFromComponent(child, SecondaryButton)) secondaryButton = child;
    });
  }

  defineCompoundComponents();
  return (
    <StyledActionRequirement {...props}>
      {illustration && (
        <BoxIllustration aria-hidden>{illustration}</BoxIllustration>
      )}
      <Content>
        {titleAsTextDisplay ? (
          <Text.Display2 as={titleLevel}>{title}</Text.Display2>
        ) : (
          <Title as={titleLevel}>{title}</Title>
        )}
        <Text.Body1 mt="small" color="deep">
          {description}
        </Text.Body1>
        {list && <Box mt="large">{list}</Box>}
        {checkable && <Box mt="xxlarge">{checkable}</Box>}
        <Actions mt="xlarge">
          {primaryButton}
          {secondaryButton}
        </Actions>
      </Content>
    </StyledActionRequirement>
  );
}

ActionRequirement.propTypes = {
  title: oneOfType([arrayOf(node), node]).isRequired,
  children: oneOfType([arrayOf(node), node]),
  description: string.isRequired,
  checkable: oneOfType([arrayOf(node), node]),
  illustration: oneOfType([arrayOf(node), node]),
  list: oneOfType([arrayOf(node), node]),
  titleAsTextDisplay: bool,
  titleLevel: string,
};

export default ActionRequirement;
