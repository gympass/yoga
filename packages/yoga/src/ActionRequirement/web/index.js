import ActionRequirement from './ActionRequirement';
import {
  Title,
  PrimaryButton,
  SecondaryButton,
} from './ActionRequirementStyles';

const ExportActionRequirement = Object.assign(ActionRequirement, {
  Title,
  PrimaryButton,
  SecondaryButton,
});

export default ExportActionRequirement;
