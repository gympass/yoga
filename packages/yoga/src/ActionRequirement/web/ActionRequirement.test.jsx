import React from 'react';
import { render } from '@testing-library/react';
import { Dumbbell, User, Star } from '@gympass/yoga-icons';
import { Text, ThemeProvider } from '../..';
import ActionRequirement from './ActionRequirement';

function handleGetList() {
  return (
    <>
      <Text>
        <Dumbbell /> 1x/day standard access
      </Text>
      <Text mt="small">
        <User /> 4x/month 1-on-1 sessions
      </Text>
      <Text mt="small">
        <Star /> 2x/month premium classes
      </Text>
    </>
  );
}

describe('<ActionRequirement />', () => {
  it('should default match snapshot', () => {
    const { container } = render(
      <ThemeProvider>
        <ActionRequirement title="title" description="description" />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with list', () => {
    const { container } = render(
      <ThemeProvider>
        <ActionRequirement
          title="title"
          description="description"
          list={handleGetList()}
        />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with text display', () => {
    const { container } = render(
      <ThemeProvider>
        <ActionRequirement
          title="title"
          description="description"
          titleAsTextDisplay
        />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with title level 2', () => {
    const { container } = render(
      <ThemeProvider>
        <ActionRequirement
          title="title"
          description="description"
          titleAsTextDisplay
          as="h2"
        />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
