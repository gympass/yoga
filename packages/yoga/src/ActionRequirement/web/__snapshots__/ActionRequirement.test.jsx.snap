// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionRequirement /> should default match snapshot 1`] = `
.c3 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  margin-top: 16px;
  color: #6B6B78;
  font-size: 16px;
}

.c2 {
  margin: 0;
  padding: 0;
  font-size: 24px;
  line-height: 32px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
  margin-bottom: 6px;
}

.c4 {
  margin-top: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:769px) {
  .c2 {
    max-width: 700px;
    font-size: 48px;
    line-height: 48px;
  }
}

@media (min-width:769px) {
  .c4 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    margin-top: 32px;
  }
}

@media (min-width:0px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:1024px) {
  .c0 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }
}

@media (min-width:0px) {
  .c1 {
    width: auto;
    margin-right: 0;
  }
}

@media (min-width:1024px) {
  .c1 {
    width: 520px;
    margin-right: 120px;
  }
}

<div>
  <div
    class="c0"
    title="title"
  >
    <div
      class="c1"
    >
      <p
        class="c2"
      >
        title
      </p>
      <p
        class="c3"
        color="deep"
        font-size="medium"
      >
        description
      </p>
      <div
        class="c4"
      />
    </div>
  </div>
</div>
`;

exports[`<ActionRequirement /> should match snapshot with list 1`] = `
.c3 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  margin-top: 16px;
  color: #6B6B78;
  font-size: 16px;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  margin-top: 24px;
  color: #6B6B78;
  font-size: 16px;
}

.c5 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  font-size: 16px;
}

.c6 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  margin-top: 16px;
  font-size: 16px;
}

.c2 {
  margin: 0;
  padding: 0;
  font-size: 24px;
  line-height: 32px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
  margin-bottom: 6px;
}

.c7 {
  margin-top: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:769px) {
  .c2 {
    max-width: 700px;
    font-size: 48px;
    line-height: 48px;
  }
}

@media (min-width:769px) {
  .c7 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    margin-top: 32px;
  }
}

@media (min-width:0px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:1024px) {
  .c0 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }
}

@media (min-width:0px) {
  .c1 {
    width: auto;
    margin-right: 0;
  }
}

@media (min-width:1024px) {
  .c1 {
    width: 520px;
    margin-right: 120px;
  }
}

<div>
  <div
    class="c0"
    list="[object Object]"
    title="title"
  >
    <div
      class="c1"
    >
      <p
        class="c2"
      >
        title
      </p>
      <p
        class="c3"
        color="deep"
        font-size="medium"
      >
        description
      </p>
      <p
        class="c4"
        color="deep"
        font-size="medium"
      >
        <div
          class=""
        >
          <p
            class="c5"
            font-size="medium"
          >
            <svg />
             1x/day standard access
          </p>
          <p
            class="c6"
            font-size="medium"
          >
            <svg />
             4x/month 1-on-1 sessions
          </p>
          <p
            class="c6"
            font-size="medium"
          >
            <svg />
             2x/month premium classes
          </p>
        </div>
      </p>
      <div
        class="c7"
      />
    </div>
  </div>
</div>
`;
